netty/src/main/java/io/grpc/netty/NettyServerHandler.java

```diff
@@ -526,6 +553,34 @@ public void onRstStreamRead(ChannelHandlerContext ctx, int streamId, long errorC
     }
 
     @Override
+    public void onPingRead(ChannelHandlerContext ctx, ByteBuf data) throws Http2Exception {
+      if (keepAliveManager != null) {
+        keepAliveManager.onDataReceived();
+      }
+      if (!keepAliveEnforcer.pingAcceptable()) {
+        ByteBuf debugData = ByteBufUtil.writeAscii(ctx.alloc(), "too_many_pings");
+        goAway(ctx, connection().remote().lastStreamCreated(), Http2Error.ENHANCE_YOUR_CALM.code(),
+            debugData, ctx.newPromise());
+        flush(ctx);
```

Comment:
So looking at it some more, it could be `ctx.flush()`. But I think I'd favor using `flush(ctx)` everywhere instead of `ctx.flush()`. Unfortunately, `flush(ctx)` sucks because it `throws Exception`, and so could be considerably more painful in situations.